src/engine/srv.c

```diff
@@ -528,7 +628,7 @@ dss_srv_handler(void *arg)
 tse_fini:
 	tse_sched_fini(&dx->dx_sched_dsc);
 crt_destroy:
-	if (dx->dx_comm)
+	if (has_crt_context(dx))
```

Comment:
I'm not sure how can that happen. For 'dx->dx_comm' case, we goto 'crt_destroy' only after 'dmi_ctx' initialized, otherwise, it'll goto 'tls_fini'. We do the same for 'xs_type == DSS_XS_SEC' case, so I think crt_context_destroy() won't be called unless the 'dmi_ctx' is initialized, right?
Please be aware that a xstream either has primary context (dx->dx_comm) or secondary context (xs_type == DSS_XS_SEC), it can't have both contexts.

Reply:
Ah, you mean the case (xs_type == DSS_XS_SEC) is not possible now (ASSERT in line 448). In this case it's fine.